kip-0042.md

```diff
@@ -0,0 +1,135 @@
+---
+KIP: '0042'
+Title: kadena_disconnect_v1 API
+Author: Danillo Felixdaal and Bart Huijgen
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2025-02-17
+---
+
+# kadena_disconnect_v1 API
+
+## Abstract
+
+This KIP standardizes the `kadena_disconnect_v1` method for terminating connections between decentralized applications (dApps) and Kadena wallets. The method securely ends active sessions, revokes permissions, and clears cached session data, ensuring a predictable and secure disconnection process.
+
+## Motivation
+
+A consistent disconnection method is crucial to protect user data and prevent unauthorized access after a session ends. By standardizing `kadena_disconnect_v1`, we aim to:
+
+- Revoke active sessions and clear cached credentials.
+- Ensure that no stale session data remains post-disconnection.
+- Provide a uniform interface for disconnecting across wallet implementations.
+- Enhance overall security and user experience.
+
+## Specification
+
+### Method Definition
+
+**Method Name:** `kadena_disconnect_v1`
+
+### Request Parameters
+
+```typescript
+interface DisconnectRequest {
+  // TODO: not sure if networkId makes sense here, ecko uses it.
```

Comment:
I wouldn't include networkId. I think you don't want to disconnect for a single network.